lib/midi_generator.rb

```diff
@@ -0,0 +1,135 @@
+# Converts Inevitable Cacophony internal note representation
+# into MIDI messages usable by an external synthesizer.
+# Based on examples in the `midilib` gem.
+
+require 'midilib/sequence'
+require 'midilib/consts'
+
+class MidiGenerator
+
+        # Middle A in MIDI
+        MIDI_TONIC = 69
+
+        # Add a phrase to the MIDI output we will generate.
+        def add_phrase(phrase)
+                @phrases ||= []
+                @phrases << phrase
+        end
+
+        # @return [Midi::Track] Notes to be output to MIDI; mainly for testing.
+        def notes_track
+                build_notes_track(sequence, @phrases)
+        end
+
+        # Write MIDI output to the given stream.
+        def write(io)
+                sequence.tracks << notes_track
```

Comment:
This isn't idempotent; if we call `write()` again, it will add the sequence to the track a second time.